tests/e2e/volumecreate_shared_test.go

```diff
@@ -0,0 +1,220 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This test is going to cover various volume creation test cases
+
+// +build runonceshared
+
+package e2e
+
+import (
+	"strings"
+	"sync"
+
+	dockerclicon "github.com/vmware/docker-volume-vsphere/tests/constants/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/misc"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+
+	. "gopkg.in/check.v1"
+)
+
+type VolumeCreateSharedTestSuite struct {
+	config     *inputparams.TestConfig
+	volumeList []string
+}
+
+var (
```

Comment:
Any reason why invalid names are a list while valid name list is returned by a function?